iothub.class.nut

```diff
@@ -5,724 +5,1024 @@
 
-    static function Parse(connectionString) {
+    ConnectionString = class {
 
-        local cn = {};
+        static function Parse(connectionString) {
+            local cn = {};
 
-        // Clean up the string first
-        if (typeof connectionString != "string") return cn;
-        connectionString = strip(connectionString);
-        if (connectionString.len() == 0) return cn;
+            // Clean up the string first
+            if (typeof connectionString != "string") return cn;
+            connectionString = strip(connectionString);
+            if (connectionString.len() == 0) return cn;
 
-        // Locate each = and ; in the connection string
-        local pairs = split(connectionString, ";")
-        foreach (pair in pairs) {
-            local kv = split(pair, "=");
-            if (kv.len() == 1) {
-                cn[kv[0]] <- null;
-            } else if (kv.len() == 2) {
-                cn[kv[0]] <- kv[1];
-                if (kv[0] == "SharedAccessKey") {
-                    cn[kv[0]] += "=";
+            // Locate each = and ; in the connection string
+            local pairs = split(connectionString, ";")
+            foreach (pair in pairs) {
+                local kv = split(pair, "=");
+                if (kv.len() == 1) {
+                    cn[kv[0]] <- null;
+                } else if (kv.len() == 2) {
+                    cn[kv[0]] <- kv[1];
+                    if (kv[0] == "SharedAccessKey") {
+                        cn[kv[0]] += "=";
+                    }
                 }
             }
-        }
 
-        // Make sure we have these fields so we don't have to check for them later
-        foreach (mandatory in ["HostName", "DeviceId"]) {
-            if (!(mandatory in cn)) {
-                cn[mandatory] <- null;
+            // Make sure we have these fields so we don't have to check for them later
+            foreach (mandatory in ["HostName", "DeviceId"]) {
+                if (!(mandatory in cn)) {
+                    cn[mandatory] <- null;
+                }
             }
+
+            return cn;
         }
 
-        return cn;
     }
-}
 
+    Authorization = class {
 
-//------------------------------------------------------------------------------
-class iothub.Endpoint {
+        static function aMinuteFromNow() {
+            return time() + 60;
+        }
 
-    static function devicePath(id) {
-        return "/devices/" + id;
-    }
+        static function fifteenMinutesFromNow() {
+            return time() + 900;
+        }
 
-    static function eventPath(id) {
-        return devicePath(id) + "/messages/events";
-    }
+        static function anHourFromNow() {
+            return time() + 3600;
+        }
 
-    static function messagePath(id) {
-        return devicePath(id) + "/messages/devicebound";
-    }
+        static function anDayFromNow() {
+            return time() + 3600 * 24;
```

Comment:
Squirrel doesn't do constant folding so it's better to use:  `return time() + 86400; //  3600 * 24`